### PR TITLE
Add Celery and Pulse ingestion in docker-compose

### LINKED ISSUE #6179: Automatic local ingestion of data
Currently we have two Pulse ingestion scripts and three Celery workers in the Procfile. Setting local ingestion requires starting the `backend` Docker container three times following [manual set up steps](https://treeherder.readthedocs.io/installation.html#running-the-ingestion-tasks).

I would like two more entries in the `docker-compose.yml` file for `pulse` and another one for the `celery` worker. They would each have their own different entryfiles.

### PR DESCRIPTION
**Pull request by imskr**
Closes #6179 

- Entries in the `docker-compose.yml` file for `pulse` and another one for the `celery worker`.

### CODE REVIEW

**Comment by armenzg** [comment]
After running `docker-compose up` I get the failure at the end.
This is why the backend has `entrypoint: './docker/entrypoint.sh'`.
It was for the MySql container to be ready.

Make sure that each one of the new containers can stand up on their own `docker-compose up <name of container>`. Right now they fail. You will need to add the appropiate environment variables.

```
celery_1      | Traceback (most recent call last):
celery_1      |   File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 220, in ensure_connection
celery_1      |     self.connect()
celery_1      |   File "/usr/local/lib/python3.7/site-packages/django/utils/asyncio.py", line 26, in inner
celery_1      |     return func(*args, **kwargs)
celery_1      |   File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 197, in connect
celery_1      |     self.connection = self.get_new_connection(conn_params)
celery_1      |   File "/usr/local/lib/python3.7/site-packages/django/utils/asyncio.py", line 26, in inner
celery_1      |     return func(*args, **kwargs)
celery_1      |   File "/usr/local/lib/python3.7/site-packages/django/db/backends/mysql/base.py", line 233, in get_new_connection
celery_1      |     return Database.connect(**conn_params)
celery_1      |   File "/usr/local/lib/python3.7/site-packages/MySQLdb/__init__.py", line 84, in Connect
celery_1      |     return Connection(*args, **kwargs)
celery_1      |   File "/usr/local/lib/python3.7/site-packages/MySQLdb/connections.py", line 179, in __init__
celery_1      |     super(Connection, self).__init__(*args, **kwargs2)
celery_1      | MySQLdb._exceptions.OperationalError: (2003, 'Can\'t connect to MySQL server on \'127.0.0.1\' (111 "Connection refused")')
```

**Comment by imskr** [comment]
@armenzg For celery, I need to give the port no. Since 3306 is occupied by SQL. which port should I use?

**Comment by armenzg** [comment]
Do you mind elaborating?

**Comment by imskr** [comment]
@armenzg I'm getting this error:
`
ERROR: for mysql  Cannot start service mysql: driver failed programming external connectivity on endpoint mysql (7c77dac98ef4ee9fe0cb827ded2fdecbd7f70633187d13da074733bd64866fdd): Error starting userland proxy: listen tcp 0.0.0.0:3306: bind: address already in use
`
So i need to change port!

**Comment by armenzg** [comment]
I don't know if you will need multiple entrypoint files. You can probably re-use the same one if all you're doing is waiting for the DB to be ready.

With the error, I would suggest you checking what port each container is using and see if there are conflicts.

**Comment by imskr** [comment]
@armenzg I have added all the required `environment` for puls-task, pulse-push, and celery. For pulse services I need to add `PULSE_URL` and according to [doc](https://treeherder.readthedocs.io/pulseload.html#1-pulse-guardian) `user` and `pass` for it.

**Comment by imskr** [comment]
@armenzg I have added all the required `environment` for puls-task, pulse-push, and celery. For pulse services I need to add `PULSE_URL` and according to [doc](https://treeherder.readthedocs.io/pulseload.html#1-pulse-guardian) `user` and `pass` for it.

File: docker-compose.yml [hunk 1]
```diff
@@ -94,6 +94,49 @@ services:
       - 'RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS=-rabbit log [{console,[{level,error}]}]'
     ports:
       - '5672:5672'
+  pulse-push:
+    build:
+      context: .
+      dockerfile: docker/dev.Dockerfile
+    environment:
+      - PULSE_URL=amqp://myuser:mypass@pulse.mozilla.org:5671/?ssl=1
```
[the remark this answers is not in the document]
Comment (armenzg) [reply docker-compose.yml:102]: Also add `PULSE_AUTO_DELETE_QUEUES=True`.

File: docker-compose.yml [hunk 2]
```diff
@@ -94,6 +94,49 @@ services:
...
+      - DATABASE_URL=mysql://root@mysql:3306/treeherder
+    entrypoint: './docker/entrypoint.sh'
+    command: ./manage.py pulse_listener_tasks
+    volumes:
+      - .:/app
+    depends_on:
+      - mysql
+      - rabbitmq
+  celery:
```
[the remark this answers is not in the document]
Comment (imskr) [reply docker-compose.yml:125]: Adding `--uid=nobody --gid=nogroup` flags removed that warning!

**Comment by imskr** [comment]
@armenzg It's working now. `BROKER_URL`  was missing in pulse services.

**Comment by armenzg** [comment]
In order to make my computer not sound like a plan about to take off and to reduce logging we will need this diff:

```diff
diff --git a/docker-compose.yml b/docker-compose.yml
index 16b5778fb..af7e1af5d 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -120,7 +120,7 @@ services:
       dockerfile: docker/dev.Dockerfile
     environment:
       - PULSE_URL=${PULSE_URL:-amqp://docker-shared-user:8r5VFxpJHtJahTVV5bYutykgDsXhGF@pulse.mozilla.org:5671/?ssl=1}
-      - LOGGING_LEVEL=DEBUG
+      - LOGGING_LEVEL=INFO
       - PULSE_AUTO_DELETE_QUEUES=True
       - DATABASE_URL=mysql://root@mysql:3306/treeherder
       - BROKER_URL=amqp://guest:guest@rabbitmq//
@@ -139,8 +139,9 @@ services:
     environment:
       - CELERY_BROKER_URL=amqp://guest:guest@rabbitmq:5672//
       - DATABASE_URL=mysql://root@mysql:3306/treeherder
+      - PROJECTS_TO_INGEST=autoland
     entrypoint: './docker/entrypoint.sh'
-    command: celery worker -A treeherder --uid=nobody --gid=nogroup --without-gossip --without-mingle --without-heartbeat -Q store_pulse_pushes,store_pulse_tasks --concurrency=3 --loglevel=INFO
+    command: celery worker -A treeherder --uid=nobody --gid=nogroup --without-gossip --without-mingle --without-heartbeat -Q store_pulse_pushes,store_pulse_tasks --concurrency=1 --loglevel=WARNING
     volumes:
       - .:/app
     depends_on:
```

**Comment by armenzg** [comment]
This is ready to take off! 🎉

**Comment by imskr** [comment]
@armenzg Thanks. Done!

**Pull request merged**